cocos/audio/audio-source.ts

```diff
@@ -341,7 +341,12 @@ export class AudioSource extends Component {
             }
         }).catch((e) => {});
     }
-
+    get sampleRate () {
+        return this._player?.sampleRate;
+    }
+    public getPCMbuffer (channelID: number): Float32Array | undefined {
+        return this._player?.getPCMBuffer(channelID);
+    }
```

Comment:
- as the `_player` is loaded asynchronously, I suggest to design the interface to return a Promise
- As they are public interface for developers, I suggest to add more detailed comment

Reply:
maybe this PR can help you https://github.com/cocos/cocos-engine/pull/11470
Or you can just cherry pick some commit here

Reply:
removed